### PR TITLE
[ODS-5191] Record Ownership Phase 2 - Update existing migration scripts to use new tables

### PR DESCRIPTION
**Pull request by raghuramakrishna**

### CODE REVIEW

**Comment by semalaiappan** [comment]
@raghuramakrishna please rebase your branch with  feature-RecordOwnershipPhase2 .  Then update table name and new columns in your script in sql server and postgres

File: Artifacts/PgSql/Data/Security/2000-AdminApp-ClaimSets.sql [hunk 1]
```diff
@@ -146,8 +157,10 @@ BEGIN
 
     RAISE NOTICE 'Updating educationStandards authorization strategy for READ.';
 
-    UPDATE dbo.resourceclaimauthorizationmetadatas
+    UPDATE dbo.ResourceClaimActionAuthorizationStrategies
     SET authorizationstrategy_authorizationstrategyid = authorization_strategy_id
+	INNER JOIN dbo.ResourceClaimActionAuthorizations
```
Comment (semalaiappan) [inline Artifacts/PgSql/Data/Security/2000-AdminApp-ClaimSets.sql:162]: update new table and columns here from this 
image

**Review by semalaiappan — CHANGES_REQUESTED** [review]

**Comment by semalaiappan** [comment]
@raghuramakrishna  bring these scripts also from old PR 
image

File: Artifacts/MsSql/Data/Security/1030-DataStandard3.3a-ResourceClaimMetadata.sql [hunk 2]
```diff
@@ -55,3 +55,15 @@
...
+
+	INSERT INTO [dbo].[ClaimSetResourceClaimActionAuthorizationStrategyOverrides]
+		([AuthorizationStrategyId]
+		,[ClaimSetResourceClaimActionId])
+	SELECT @noFurtherAuthRequiredAuthorizationStrategyId, crc.ClaimSetResourceClaimActionId
+	FROM [dbo].[ClaimSetResourceClaimActions] crc
+	INNER JOIN [dbo].[Actions] a ON
+		crc.ActionId=a.Actionid and 
+	    a.ActionName IN ('Create','Read','Update','Delete')
+	INNER JOIN [dbo].[ResourceClaims] r ON
```
Comment (semalaiappan) [inline Artifacts/MsSql/Data/Security/1030-DataStandard3.3a-ResourceClaimMetadata.sql:69]: Inner join is missing dbo.ClaimSets
    WHERE ClaimSetName IN ('SIS Vendor','Ed-Fi Sandbox','District Hosted SIS Vendor')

File: Artifacts/MsSql/Data/Security/1070-OrganizationDepartmentMetadata.sql [hunk 3]
```diff
@@ -55,16 +72,26 @@ SELECT  @schoolClaimId = ResourceClaimId
 FROM    dbo.ResourceClaims rc 
 WHERE   rc.ClaimName = 'http://ed-fi.org/ods/identity/claims/school'
 
-IF NOT EXISTS (SELECT 1 FROM dbo.ClaimSetResourceClaims csrc INNER JOIN dbo.ClaimSets cs ON csrc.ClaimSet_ClaimSetId = cs.ClaimSetId WHERE cs.ClaimSetName = 'District Hosted SIS Vendor' AND csrc.ResourceClaim_ResourceClaimId = @schoolClaimId)
```
Comment (semalaiappan) [inline Artifacts/MsSql/Data/Security/1070-OrganizationDepartmentMetadata.sql:58]: dbo.ResourceClaimAuthorizationMetadatas is old table  and new table is ResourceClaimActions  . this line needs to be updated IF NOT EXISTS (SELECT 1 FROM dbo.ResourceClaimAuthorizationMetadatas WHERE ResourceClaim_ResourceClaimId = @organizationDepartmentClaimId)

Comment (semalaiappan) [reply Artifacts/MsSql/Data/Security/1070-OrganizationDepartmentMetadata.sql:58]: INSERT INTO dbo.ResourceClaimAuthorizationMetadatas(ResourceClaim_ResourceClaimId, Action_ActionId, AuthorizationStrategy_AuthorizationStrategyId)  dbo.ResourceClaimAuthorizationMetadatas is old table and new table is ResourceClaimActions .

File: Artifacts/MsSql/Data/Security/2000-AdminApp-ClaimSets.sql [hunk 4]
```diff
@@ -55,19 +60,42 @@ SELECT @actionId = ActionId FROM dbo.Actions WHERE ActionName = @actionName
 SELECT @claimSetId = ClaimSetId FROM dbo.ClaimSets WHERE ClaimSetName = @claimSetName
```
Comment (semalaiappan) [inline Artifacts/MsSql/Data/Security/2000-AdminApp-ClaimSets.sql:60]: SELECT @actionId = ActionId FROM dbo.Actions WHERE ActionName = @actionName  . Where is set value for @actionName variable

File: Artifacts/MsSql/Data/Security/2000-AdminApp-ClaimSets.sql [hunk 5]
```diff
@@ -55,19 +60,42 @@ SELECT @actionId = ActionId FROM dbo.Actions WHERE ActionName = @actionName
...
 
 PRINT 'Configuring Claims for Ed-Fi ODS Admin App Claimset...'
-INSERT INTO dbo.ClaimSetResourceClaims
-    (Action_ActionId, ClaimSet_ClaimSetId, ResourceClaim_ResourceClaimId, AuthorizationStrategyOverride_AuthorizationStrategyId)
-SELECT ActionId, @claimSetId, ResourceClaimId , @authorizationStrategyId FROM dbo.Actions a, @resourceClaimIds rc
-WHERE NOT EXISTS (SELECT 1
-                  FROM dbo.ClaimSetResourceClaims
-				  WHERE Action_ActionId = a.ActionId AND ClaimSet_ClaimSetId = @claimSetId AND ResourceClaimId = rc.ResourceClaimId)
+IF NOT EXISTS (SELECT 1
+                  FROM dbo.ClaimSetResourceClaimActions
```
Comment (semalaiappan) [inline Artifacts/MsSql/Data/Security/2000-AdminApp-ClaimSets.sql:64]: missing with condition AND ResourceClaimId = rc.ResourceClaimId in if not exists

**Review by semalaiappan — CHANGES_REQUESTED** [review]

**Comment by actionname** [comment]
您好！我已经收到你的邮件我会尽快回复你，谢谢！

File: Artifacts/MsSql/Data/Security/2000-AdminApp-ClaimSets.sql [hunk 6]
```diff
@@ -55,19 +60,42 @@ SELECT @actionId = ActionId FROM dbo.Actions WHERE ActionName = @actionName
...
+INSERT INTO [dbo].[ClaimSetResourceClaimActionAuthorizationStrategyOverrides]
+    ([AuthorizationStrategyId]
+    ,[ClaimSetResourceClaimActionId])
+SELECT @authorizationStrategyId, csrc.ClaimSetResourceClaimActionId
+FROM [dbo].[ClaimSetResourceClaimActions] csrc
+INNER JOIN [dbo].[ResourceClaims] r ON 
+	csrc.ResourceClaimId = r.ResourceClaimId
+INNER JOIN [dbo].[Actions] a ON
+	a.actionId =csrc.ActionId
+	AND a.ActionName in ('READ')
```
Comment (semalaiappan) [inline Artifacts/MsSql/Data/Security/2000-AdminApp-ClaimSets.sql:97]: change to 'Read' even though sql server is not case sensitive language

**Review by semalaiappan — CHANGES_REQUESTED** [review]

File: Artifacts/MsSql/Data/Security/2000-AdminApp-ClaimSets.sql (hunk 5 above)
Comment (semalaiappan) [reply Artifacts/MsSql/Data/Security/2000-AdminApp-ClaimSets.sql:63]: IF NOT EXISTS (SSELECT 1
                  FROM dbo.ClaimSetResourceClaimActions csraa,dbo.Actions a, @resourceClaimIds rc
				  WHERE csraa.ActionId = a.ActionId AND ClaimSetId = @claimSetId AND csraa.ResourceClaimId = rc.ResourceClaimId)

**Review by semalaiappan — CHANGES_REQUESTED** [review]

File: Artifacts/PgSql/Data/Security/1070-OrganizationDepartmentMetadata.sql [hunk 7]
```diff
@@ -12,64 +12,98 @@ relationships_authorization_strategy_id INT;
...
-IF NOT EXISTS (SELECT 1 FROM dbo.ResourceClaimAuthorizationMetadatas WHERE ResourceClaim_ResourceClaimId = organization_department_claim_id)
-THEN
-    RAISE NOTICE 'Assigning default metadata for CRUD operations on OrganizationDepartment to the relationship-based authorization strategy.';
-    
-    INSERT INTO dbo.ResourceClaimAuthorizationMetadatas(ResourceClaim_ResourceClaimId, Action_ActionId, AuthorizationStrategy_AuthorizationStrategyId)
-    SELECT  organization_department_claim_id, ActionId, relationships_authorization_strategy_id
-    FROM    dbo.Actions
-    WHERE   ActionName IN ('Create', 'Read', 'Update', 'Delete');
-END IF;
+	IF NOT EXISTS (SELECT 1 FROM dbo.ResourceClaimAuthorizationMetadatas WHERE ResourceClaim_ResourceClaimId = organization_department_claim_id)
```
Comment (semalaiappan) [inline Artifacts/PgSql/Data/Security/1070-OrganizationDepartmentMetadata.sql:59]: ResourceClaimAuthorizationMetadatas is old table  , please update with new table . refer sql server version script

**Review by semalaiappan — CHANGES_REQUESTED** [review]

**Comment by semalaiappan** [comment]
@raghuramakrishna   Count for 4 tables should be matching for Sql server and Postgres .  Validate that in your local run for both database .

**Comment by semalaiappan** [comment]
@raghuramakrishna  ClaimSetResourceClaimActionAuthorizationStrategyOverrides table record count is mismatch with Sql server and Postgres now .

**Comment by semalaiappan** [comment]
As All 4 tables changes are looks good .   Integration builds failure will be fixed in separate ticket, So  I am merging this ticket .

**Pull request merged**